### ISSUE TITLE
Doesn't report error when `clang-format` encounters an error

### ISSUE THREAD

**Issue by bcaddy** [cpp-linter/cpp-linter#25]
When `clang-tidy` encounters an error and doesn't run then the workflow should fail. Currently it doesn't, see [this workflow run](https://github.com/bcaddy/cholla/actions/runs/3447648055/jobs/5753874789) for an example.

**Comment by 2bndy5** [comment]
I remember now why we currently don't fail fast when a clang tool returns a non-zero exit code. Clang-tidy returns an error code when a compilation database isn't found. It is pretty common for people to not provide or generate a compilation database; in which case, clang-tidy continues to process the file with default values/flags, but this often results in a "compiler error".

<details><summary>typical example output from clang-tidy</summary>

This is the output from our unit test(s) that are supposed to pass:

```
Error while trying to load a compilation database:
Could not auto-detect compilation database for file "opl/opl.c"
No compilation database found in path\to\GitHub\cpp-linter\tests\capture_tools_output\chocolate-doom\opl or any parent directory
fixed-compilation-database: Error while opening fixed database: no such file or directory
json-compilation-database: Error while opening JSON database: no such file or directory
Running without flags.
3883 warnings and 1 error generated.
Error while processing path\to\GitHub\cpp-linter\tests\capture_tools_output\chocolate-doom\opl\opl.c.
Suppressed 3883 warnings (3857 in non-user code, 26 due to line filter).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
Found compiler error(s).
```
</details>

So, I'm forced to use a default value of `false` for the new fail-fast option...

I do intend to support a finer grained control using env vars that specify if clang-format and/or clang-tidy should fail fast. This finer control has to be done with env vars because the new fail-fast input option would get passed to `capture_clang_tools_output()` which calls both `run_clang_tidy()` and then `run_clang_format()`.

Just to be perfectly clear, the cpp-linter-action could offer 2-3 input options about controlling the fail-fast behavior, but the cpp-linter CLI will only support 1 fail-fast argument. In other words, people using the pkg directly (currently most recommended) would have to set env vars for finer control:
```yaml
      - name: run cpp-linter via CLI
        env:
            FAIL_FAST_CLANG_FORMAT: true
            FAIL_FAST_CLANG_TIDY: false
        # both env vars could also be set simultaneously with a CLI option
        run: cpp-linter --fail-fast=true
```
But people using the action could use a newer set of input options that get implemented as env vars under the hood:
```yaml
      - name: run cpp-linter via github actions
        uses: cpp-linter/cpp-linter-action
        with:
            fail-fast-clang-format: true
            fail-fast-clang-tidy: false
            # or set both with one option:
            fail-fast: true
```
